From 6a443a73b796fa0a6e52b98751991a2149310f85 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Mon, 28 Mar 2022 18:23:29 +0200
Subject: [PATCH 066/120] remoteproc: move COPRO state definition in driver

The definitions are use only by the stm32_rproc driver.
No need to export them in stm32.h

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git db1e75eb4683e9a20c2ae0b7677b8a1f891e279b remoteproc: move COPRO state definition in driver]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: If486476c61661f91c6268fcac4ebdafd3bd91ab2
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/315249
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Michel JAOUEN <michel.jaouen@st.com>
---
 arch/arm/mach-stm32mp/include/mach/stm32.h | 6 ------
 drivers/remoteproc/stm32_copro.c           | 8 ++++++++
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-stm32mp/include/mach/stm32.h b/arch/arm/mach-stm32mp/include/mach/stm32.h
index 41eed82cfa5..9bce42fef09 100644
--- a/arch/arm/mach-stm32mp/include/mach/stm32.h
+++ b/arch/arm/mach-stm32mp/include/mach/stm32.h
@@ -144,12 +144,6 @@ enum forced_boot_mode {
 #define TAMP_FWU_BOOT_IDX_MASK		GENMASK(3, 0)
 
 #define TAMP_FWU_BOOT_IDX_OFFSET	0
-#define TAMP_COPRO_STATE_OFF		0
-#define TAMP_COPRO_STATE_INIT		1
-#define TAMP_COPRO_STATE_CRUN		2
-#define TAMP_COPRO_STATE_CSTOP		3
-#define TAMP_COPRO_STATE_STANDBY	4
-#define TAMP_COPRO_STATE_CRASH		5
 #endif
 
 #ifdef CONFIG_STM32MP13X
diff --git a/drivers/remoteproc/stm32_copro.c b/drivers/remoteproc/stm32_copro.c
index 28951b5c2d4..ea084710339 100644
--- a/drivers/remoteproc/stm32_copro.c
+++ b/drivers/remoteproc/stm32_copro.c
@@ -18,6 +18,14 @@
 
 #define STM32MP15_M4_FW_ID 0
 
+/* TAMP_COPRO_STATE register values */
+#define TAMP_COPRO_STATE_OFF		0
+#define TAMP_COPRO_STATE_INIT		1
+#define TAMP_COPRO_STATE_CRUN		2
+#define TAMP_COPRO_STATE_CSTOP		3
+#define TAMP_COPRO_STATE_STANDBY	4
+#define TAMP_COPRO_STATE_CRASH		5
+
 /**
  * struct stm32_copro_privdata - power processor private data
  * @reset_ctl:		reset controller handle
-- 
2.47.2

